src/main/java/com/yelp/nrtsearch/server/luceneserver/search/TerminateAfterWrapper.java

```diff
@@ -130,14 +134,30 @@ public TerminateAfterLeafCollectorWrapper(LeafCollector leafCollector) {
 
       @Override
       public void setScorer(Scorable scorer) throws IOException {
-        leafCollector.setScorer(scorer);
+        if (terminatedEarly) {
+          // setup constant scorer to avoid spending time on scoring
+          float score = 1;
```

Comment:
the `1` is temporary, we need to finalize a value here in order the constant scores not to impact the collected hits.

Reply:
Yes. You should try to find the min score supported by lucene and use the min score.

Reply:
Yup, sounds good, will update accordingly

Reply:
Updated